File: src/main/java/javaCode/ProductsModel.java
package javaCode;

public record ProductsModel(
        int id,
        boolean isActive,
        boolean isArchived,
        String description,
        String name,
        String nameFull,
        int price,
        int bonuses,
        int weight,
        ProductSaldo productSaldo) {
    public record ProductSaldo(String productCode,
                               String scladId,
                               int volume,
                               int regionId) {
    }

}

File: src/main/java/kotlinCode/ProductsModel.kt
package kotlin

data class ProductsModel(
    val id: Int,
    val isActive: Boolean,
    val isArchived: Boolean,
    val description: String,
    val name: String,
    val nameFull: String,
    val price: Int,
    val bonuses: Int,
    val weight: Int,
    val productSaldo: ProductSaldo
) {
    data class ProductSaldo(
        val productCode: String,
        val scladId: String,
        val volume: Int,
        val regionId: Int
    )


}